receiver/dcgmreceiver/client_test.go

```diff
@@ -12,42 +12,23 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build gpu
-// +build gpu
+//go:build gpu && !has_gpu
+// +build gpu,!has_gpu
 
 package dcgmreceiver
 
 import (
 	"errors"
-	"fmt"
-	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-	"go.uber.org/zap"
-	"go.uber.org/zap/zapcore"
 	"go.uber.org/zap/zaptest"
 )
 
 func TestNewDcgmClientOnInitializationError(t *testing.T) {
-	realDcgmInit := dcgmInit
-	defer func() { dcgmInit = realDcgmInit }()
-	dcgmInit = func(args ...string) (func(), error) {
-		return nil, fmt.Errorf("No DCGM client library *OR* No DCGM connection")
-	}
-
-	seenDcgmConnectionWarning := false
-	logger := zaptest.NewLogger(t, zaptest.WrapOptions(zap.Hooks(func(e zapcore.Entry) error {
-		if e.Level == zap.WarnLevel && strings.Contains(e.Message, "Unable to connect to DCGM daemon") {
```

Comment:
I'm not sure we wanted to remove the test for this failure mode. There are two ways creating a new client could fail, right? Or am I missing something?

Reply:
Good point! Actually it looks https://github.com/NVIDIA/go-dcgm/issues/13 just got fixed. With this fix, we should not need to check the `.so` file in our code anymore. Let me upgrade and remove the temporary `isDcgmInstalled()` function.